cmd/bundle/exec.go

```diff
@@ -0,0 +1,152 @@
+package bundle
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/databricks/cli/bundle/config/mutator"
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/spf13/cobra"
+)
+
+func newExecCommand() *cobra.Command {
+	execCmd := &cobra.Command{
+		Use:   "exec",
+		Short: "Execute a command using the same authentication context as the bundle",
+		Args:  cobra.MinimumNArgs(1),
+		Long: `Execute a command using the same authentication context as the bundle
+
+The current working directory of the provided command will be set to the root
+of the bundle.
+
+Example usage:
+1. databricks bundle exec -- echo "hello, world"
+2. databricks bundle exec -- /bin/bash -c "echo hello"
+3. databricks bundle exec -- uv run pytest`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.ArgsLenAtDash() != 0 {
+				return fmt.Errorf("Please add a '--' separator. Usage: 'databricks bundle exec -- %s'", strings.Join(args, " "))
+			}
+
+			// Load the bundle configuration to get the authentication credentials.
+			b, diags := root.MustConfigureBundle(cmd)
+			if diags.HasError() {
+				return diags.Error()
+			}
+
+			childCmd := exec.Command(args[0], args[1:]...)
```

Comment:
this might panic if only 1 arg is provided

Reply:
No it's fine. We have an acceptance test that tests this `$CLI bundle exec -- pwd`.